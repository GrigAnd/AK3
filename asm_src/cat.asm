_start:
        LD  4343
        ST  4242
        JMP _start