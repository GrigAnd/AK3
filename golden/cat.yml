in_source: |-
  _start:
        LD  4343
        ST  4242
        JMP _start
in_stdin: |-
  hello world
out_code: |-
  [{"position": 0, "opcode": "LD", "operand": 4343, "op_type": "DIRECT", "src_line": 2},
  {"position": 1, "opcode": "ST", "operand": 4242, "op_type": "DIRECT", "src_line": 3},
  {"position": 2, "opcode": "JMP", "operand": 0, "op_type": "DIRECT", "src_line": 4}]
out_stdout: |
  source LoC: 4, code instrs: 3, data: 0
  ============================================================
  hello world
  Instructions executed: 33
out_log: |
  DEBUG   machine:run           TICK:   1 PC:   1 ACC: 104 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:   2 PC:   2 ACC: 104 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:   2 PC:   0 ACC: 104 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:   3 PC:   1 ACC: 101 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:   4 PC:   2 ACC: 101 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:   4 PC:   0 ACC: 101 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:   5 PC:   1 ACC: 108 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:   6 PC:   2 ACC: 108 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:   6 PC:   0 ACC: 108 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:   7 PC:   1 ACC: 108 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:   8 PC:   2 ACC: 108 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:   8 PC:   0 ACC: 108 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:   9 PC:   1 ACC: 111 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  10 PC:   2 ACC: 111 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  10 PC:   0 ACC: 111 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:  11 PC:   1 ACC:  32 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  12 PC:   2 ACC:  32 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  12 PC:   0 ACC:  32 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:  13 PC:   1 ACC: 119 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  14 PC:   2 ACC: 119 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  14 PC:   0 ACC: 119 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:  15 PC:   1 ACC: 111 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  16 PC:   2 ACC: 111 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  16 PC:   0 ACC: 111 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:  17 PC:   1 ACC: 114 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  18 PC:   2 ACC: 114 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  18 PC:   0 ACC: 114 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:  19 PC:   1 ACC: 108 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  20 PC:   2 ACC: 108 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  20 PC:   0 ACC: 108 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           TICK:  21 PC:   1 ACC: 100 MEM: N/A ADDR: 4343 	LD
  DEBUG   machine:run           TICK:  22 PC:   2 ACC: 100 MEM: N/A ADDR: 4242 	ST
  DEBUG   machine:run           TICK:  22 PC:   0 ACC: 100 MEM: N/A ADDR: 4242 	JMP
  DEBUG   machine:run           EOF
out_data: '[]'
